src/binder/assemblybinder.cpp

```diff
@@ -1158,13 +1165,48 @@ namespace BINDER_SPACE
                                                                 pBindResult));
             }
         }
-        else
+        else 
         {
-            // Is assembly on TPA list?
+            ReleaseHolder<Assembly> pTPAAssembly;
             SString &simpleName = pRequestedAssemblyName->GetSimpleName();
+
+            // Is assembly in the bundle?
+            // Single-file bundle contents take precedence over TPA.
+            // Bundled assemblies are treated similar to TPAs.
+            if (BundleInfo::AppIsBundle())
+            {
+                SString candidates[] = { W(".dll"), W(".ni.dll") };
+
+                // Loop through the binding paths looking for a matching assembly
+                for (DWORD i = 0; i < 2; i++)
+                {
+                    SString bundledName(simpleName);
+                    bundledName.Append(candidates[i]);
```

Comment:
Nit: please rename the `bundleName` to probably just `assemblyFileName` or something similar - it is NOT the name of the bundle itself - confusing.